koku/api/report/queries.py

```diff
@@ -315,6 +318,29 @@ def _get_search_filter(self, filters):  # noqa C901
         LOG.debug(f"self.query_exclusions: {self.query_exclusions}")
         return composed_filters
 
+    def _provider_map_conditional_exclusions(self):
+        """
+        Uses the provider_map conditional_filter to exclude from a query in certain scenarios.
+
+        Such as when we fall back to the daily summary table but don't want Unallocated projects
+        included for OCP compute/memory endpoints.
+        """
+
+        exclusions = QueryFilterCollection()
+        conditional_filter = self._mapper.report_type_map.get("conditional_filter")
+        if (
+            conditional_filter
+            and conditional_filter.get("exclude")
+            and self.query_table == conditional_filter.get("if_table")
+        ):
+            to_exclude = conditional_filter.get("exclude")
+            if isinstance(to_exclude, list):
+                for exclusion in to_exclude:
+                    exclusions.add(**exclusion)
+            else:
+                exclusions.add(**to_exclude)
+        return exclusions.compose()
```

Comment:
Almost all of this branching can be eliminated if the above suggestion is applied.